data-science-onramp/data-ingestion/setup_test.py

```
@@ -0,0 +1,170 @@
+BUCKET_NAME = f'setup-test-code-{uuid.uuid4()}'
+DESTINATION_BLOB_NAME = "setup.py"
+JOB_FILE_NAME = f'gs://{BUCKET_NAME}/setup.py'
+JOB_DETAILS = {  # Job configuration
+    'placement': {
+        'cluster_name': CLUSTER_NAME
+    },
+    'pyspark_job': {
+        'main_python_file_uri': JOB_FILE_NAME,
+        'args': [
+            BUCKET_NAME,
+            "--test",
+        ],
+        "jar_file_uris": [
+                "gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar"
+        ],
+    },
+}
+CLUSTER_DATA = {  # Create cluster configuration
+    'project_id': PROJECT,
+    'cluster_name': CLUSTER_NAME,
+    'config': {
+        'gce_cluster_config': {
+            'zone_uri': '',
+            "metadata": {
+                "PIP_PACKAGES": "google-cloud-storage"
+            },
+        },
+        'master_config': {
+            'num_instances': 1,
+            'machine_type_uri': 'n1-standard-8'
+        },
+        'worker_config': {
+            'num_instances': 6,
+            'machine_type_uri': 'n1-standard-8'
+        },
+        "initialization_actions": [
+            {
+                "executable_file": ("gs://dataproc-initialization-actions/"
+                                    "python/pip-install.sh"),
+            }
+        ],
+        "software_config": {
+            "image_version": "1.5.4-debian10",
+            "optional_components": [
+                "ANACONDA"
+            ],
+        }
+    }
+}
+
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_cluster():
+    # Create cluster using cluster client
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': f'{REGION}-dataproc.googleapis.com:443'
+    })
+    operation = cluster_client.create_cluster(PROJECT, REGION, CLUSTER_DATA)
+
+    # Wait for cluster to provision
+    operation.result()
+
+    yield
+
+    # Delete cluster
+    operation = cluster_client.delete_cluster(PROJECT, REGION,
```

What happens if the cluster isn't found? You made need to add a try/except here. @bradmiro plz chime in if there's a best practice

---

I typically just let this fail loudly. In any instance where the cluster isn't properly deleted, it is usually indicative of other problems.